integration-tests/src/test/java/org/zowe/apiml/functional/cloudgateway/CentralRegistryTest.java

```diff
@@ -0,0 +1,123 @@
+import org.junit.jupiter.api.Test;
+import org.springframework.http.HttpStatus;
+import org.zowe.apiml.product.constants.CoreService;
+import org.zowe.apiml.util.TestWithStartedInstances;
+import org.zowe.apiml.util.categories.DiscoverableClientDependentTest;
+import org.zowe.apiml.util.config.CloudGatewayConfiguration;
+import org.zowe.apiml.util.config.ConfigReader;
+import org.zowe.apiml.util.config.SslContext;
+import org.zowe.apiml.util.config.SslContextConfigurer;
+import org.zowe.apiml.util.config.TlsConfiguration;
+
+import java.net.URI;
+import java.net.URL;
+import java.util.List;
+import java.util.Map;
+
+import static com.google.common.base.Strings.nullToEmpty;
+import static io.restassured.RestAssured.given;
+import static io.restassured.RestAssured.with;
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.hasSize;
+import static org.hamcrest.Matchers.is;
+
+@DiscoverableClientDependentTest
+@Tag("CloudGatewayCentralRegistry")
+class CentralRegistryTest implements TestWithStartedInstances {
+    static final String CENTRAL_REGISTRY_PATH = "/" + CoreService.CLOUD_GATEWAY.getServiceId() + "/api/v1/registry/";
+
+    static CloudGatewayConfiguration conf = ConfigReader.environmentConfiguration().getCloudGatewayConfiguration();
+
+    @BeforeAll
+    @SneakyThrows
+    static void setupAll() {
+        //In order to avoid config customization
+        ConfigReader.environmentConfiguration().getGatewayServiceConfiguration().setInstances(2);
+
+        TlsConfiguration tlsCfg = ConfigReader.environmentConfiguration().getTlsConfiguration();
+        SslContextConfigurer sslContextConfigurer = new SslContextConfigurer(tlsCfg.getKeyStorePassword(), tlsCfg.getClientKeystore(), tlsCfg.getKeyStore());
+        SslContext.prepareSslAuthentication(sslContextConfigurer);
+    }
+
+    @BeforeEach
+    void setup() {
+        RestAssured.useRelaxedHTTPSValidation();
+    }
+
+    @Test
+    @SneakyThrows
+    void shouldFindRegisteredGatewayInCentralApiml() {
+        ValidatableResponse response = listCentralRegistry("domain-apiml", "zowe.apiml.gateway", null);
+
+        List<Map<String, Object>> services = response.extract().jsonPath().getObject("[0].services", new TypeRef<List<Map<String, Object>>>() {
+        });
+
+        assertThat(services, hasSize(1));
+    }
+
+    @Test
+    void shouldFindTwoApimlInstancesInTheCentralRegistry() {
+        listCentralRegistry(null, null, null)
+            .body("size()", is(2));
+    }
+
+    @Test
+    void shouldFindBothApimlIds() {
+        List<String> apimlIds = listCentralRegistry(null, null, null)
+            .extract().jsonPath().getList("apimlId");
+
+        assertThat(apimlIds, Matchers.hasItems(Matchers.equalTo("domain-apiml"), Matchers.equalTo("node-apiml")));
+    }
+
+    @Test
+        //@Disabled("This test should be enabled after the x509 projection is implemented")
+    void shouldRejectUnauthorizedAccessToCentralRegistry() {
```

Comment:
Can you add a test where you provide the client cert for an unauthorized user?

Reply:
`shouldRejectUntrustedX509CertificateToAccessCentralRegistry` has been added